src/dfx/src/lib/builders/mod.rs

```diff
@@ -180,26 +180,86 @@ pub trait CanisterBuilder {
                 .context("Failed to read language bindings archive entries.")?
             {
                 let mut file = f.context("Failed to read language bindings archive entry.")?;
-                let mut file_contents = String::new();
-                file.read_to_string(&mut file_contents)
-                    .context("Failed to read language bindings archive file content.")?;
-
-                let mut new_file_contents = file_contents
-                    .replace("{canister_id}", &info.get_canister_id()?.to_text())
-                    .replace("{canister_name}", info.get_name());
-                new_file_contents = match &info.get_declarations_config().env_override {
-                    Some(s) => new_file_contents.replace(
-                        "process.env.{canister_name_uppercase}_CANISTER_ID",
-                        &format!("\"{}\"", s),
-                    ),
-                    None => new_file_contents
-                        .replace("{canister_name_uppercase}", &info.get_name().to_uppercase()),
-                };
-                let index_js_path = generate_output_dir.join("index").with_extension("js");
-                std::fs::write(&index_js_path, new_file_contents).with_context(|| {
-                    format!("Failed to write to {}.", index_js_path.to_string_lossy())
-                })?;
-                eprintln!("  {}", &index_js_path.display());
+
+                let pathname: PathBuf = file
+                    .path()
+                    .context("Failed to read language bindings entry path name.")?
+                    .to_path_buf();
+                let extension = pathname.extension();
+                let is_template = matches! (extension, Some (ext ) if ext == OsStr::new("hbs"));
+
+                if is_template {
+                    let mut file_contents = String::new();
+                    file.read_to_string(&mut file_contents)
+                        .context("Failed to read language bindings archive file content.")?;
+
+                    // create the handlebars registry
+                    let handlebars = Handlebars::new();
+
+                    let mut data = BTreeMap::new();
+
+                    let canister_name = &info.get_name().to_string();
+
+                    let node_compatibility =
+                        match &info.get_declarations_config().node_compatibility {
+                            Some(s) => s.to_owned(),
+                            None => false,
+                        };
+
+                    // Insert only if node outputs are specified
+                    let mut node_requirements = String::new();
+
+                    let mut actor_export =
+                        format!("export const {} = createActor(canisterId);", canister_name)
+                            .to_string();
+
+                    if node_compatibility {
+                        node_requirements = "import fetch from \"isomorphic-fetch\"".to_string();
```

Comment:
also, can this template string not be a file that gets read instead?

Reply:
This can be a file, but there is no significant benefit to it, and comes with extra overhead. the Rust handlebars implementation is less robust than the JS one